binding/python/docs/TUTORIAL.md

```diff
@@ -0,0 +1,144 @@
+# How to write python code with multiverso
+1. You could start with the [test example](../multiverso/tests/test_multiverso.py) to learn the basic use of multiverso apis.
+2. After understanding the basic use of multiverso apis, you can test and run the [examples](../examples/).  The original code links are written at the beginning of them. You can compare the multiverso version with the original ones to find the differences. The places need to be modified to use multiverso are all inserted comments like `# MULTIVERSO: XXX`
+
+Here is a typical example of python code with multvierso.
+```python
+# import multiverso.
+import multiverso as mv
+# Init multiverso.
+mv.init()
+# Get total number of workers.
+print mv.workers_num()
+# Get the id for current worker.
+print mv.worker_id()
+# if the worker is master worker.
+print mv.is_master_worker()
+
+# Here is your code to create tables, train models and sync values.
+
+# Shutdown multiverso
+mv.shutdown()
+```
+
+## About the master worker
+Some things only need one worker process, such as validation, outputting the result, initializing the parameters and so on. So we mark the worker 0 as the master worker to finish these things. You can use `mv.is_master_worker` to distinguish the master worker from others.
+For example, if you want to make sure only one process will initialize the parameters, you can write similar code below.
+```
+import multiverso as mv
+# create your table handler tbh and initial value params
+if mv.is_master_worker():
+    # Only master worker will set the initial value.
+    tbh.add(params)
+    # Set a barrier for other workers to wait.
+    mv.barrier()
+else:
+    # Wait the master worker to finish setting.
+    mv.barrier()
+    # Get the initial model from the server.
+    params = tbh.get()
+```
+
+There are similar strategies in `theano_ext.sharedvar` and `lasagne_ext.param_manager` when initializing values. They are already implemented in the constructors.
```

Comment:
Similar strategies also apply to `theano_ext.sharedvar` and `lasagne_ext.param_manager` during initialization and it's already implemented in the constructors.